internal/util/update/resource-merge.go

```diff
@@ -242,6 +240,17 @@ func getSubDirsAndNonKrmFiles(root string) (sets.String, sets.String, error) {
 	return dirs, files, nil
 }
 
+func shouldSkipFile(matchFilesGlob []string, path string) (bool, error) {
```

Comment:
Should we add a test for this.

Reply:
Why do we need to re-implement the logic here ? Why can't we use the shouldSkipFile method in kyaml ?

Reply:
So we can try to reuse code, but the way it was done here is not the way to do it. We can either use the `LocalPackageReader` properly, or we can split out the code to check if a file matches a slice of globs from the `LocalPackageReader`. I don't think the former is doable, and I'm not sure if this is enough code that it is worth creating a separate function in kyaml.

I made an update to the tests, so we also verify that we don't handle krmfiles in the `ReplaceNonKRMFiles` function.